Add Token Caching Identity documentation. (#34389)
Azure/azure-sdk-for-java: sdk/identity/azure-identity/README.md

- [Azure Virtual Machines](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token)
- [Azure Virtual Machines Scale Sets](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/qs-configure-powershell-windows-vmss)

> **Note:** Use `azure-identity` version `1.7.0` or later to utilize in memory token caching support for managed identity authentication.

Reviewer: ```suggestion
**Note:** Use `azure-identity` version `1.7.0` or later to utilize in-memory token caching support for managed identity authentication.
```

> **Note:** Use `azure-identity` version `1.7.0` or later to utilize in-memory token caching support for managed identity authentication.

Reviewer: The "Azure App Service and Azure Functions" link above should use `java` instead of `dotnet` in the querystring.

Also, let's link to the token caching doc from this sentence.

```suggestion
**Note:** Use `azure-identity` version `1.7.0` or later to utilize [token caching](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md) support for managed identity authentication.
```

Configuration is attempted in the above order. For example, if values for a client secret and certificate are both present, the client secret will be used.

## Token Caching

> Token caching is a feature provided by the Azure Identity client library that allows applications to cache tokens and improve their resilience, performance, and reduce the number of requests made to Azure Active Directory (Azure AD) to obtain access tokens.

Reviewer: ```suggestion
Token caching is a feature provided by the Azure Identity library that allows apps to:

- Cache tokens.
- Improve resilience and performance.
- Reduce the number of requests made to Azure Active Directory (Azure AD) to obtain access tokens.
```

> The Azure Identity client library offers both in-memory caching and persistent disk caching. For more details refer to [token caching documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md)

Reviewer: ```suggestion
The Azure Identity library offers both in-memory and persistent disk caching. For more details, see the [token caching documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md)
```

Configuration is attempted in the above order. For example, if values for a client secret and certificate are both present, the client secret will be used.

> ## Token Caching

Reviewer: ```suggestion
## Token caching

```

Token caching is a feature provided by the Azure Identity library that allows apps to:

> - Cache tokens.

Reviewer: ```suggestion
- Cache tokens in memory (default) or on disk (opt-in).
```

- Improve resilience and performance.

> - Reduce the number of requests made to Azure Active Directory (Azure AD) to obtain access tokens.

Reviewer: ```suggestion
- Reduce the number of requests made to Azure AD to obtain access tokens.
```

> The Azure Identity library offers both in-memory and persistent disk caching. For more details, see the [token caching documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md)

Reviewer: ```suggestion
The Azure Identity library offers both in-memory and persistent disk caching. For more details, see the [token caching documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/TOKEN_CACHING.md).
```
